File: Python/metaSDT/_options.py
"""Default option helpers for Python shell functions."""

import warnings


def default_shell_run_m_option():
    """Return default shell-level options for :func:`shell_run_m`."""
    return {
        "n": 1000,
        "seed": None,
        "plot": True,
        "density_points": 512,
        "xlim": None,
        "show": True,
    }


def merge_shell_run_m_option(option=None):
    if option is None:
        option = {}
    if not isinstance(option, dict):
        raise TypeError("option must be a dict or None.")

    merged_option = default_shell_run_m_option()
    unknown_keys = sorted(set(option.keys()) - set(merged_option.keys()))
    if unknown_keys:
        warnings.warn(
            "Unknown shell_run_m option field(s): "
            + ", ".join(str(key) for key in unknown_keys),
            RuntimeWarning,
            stacklevel=2,
        )
    merged_option.update(option)
    return merged_option

File: R/R/options.R
#' Default options for shell_run_m
#'
#' @return A list of shell-level simulation and display options.
#' @export
default_shell_run_m_option <- function() {
    list(
        n = 1000L,
        seed = NULL,
        plot = TRUE,
        density_points = 512L,
        xlim = NULL,
        show = TRUE
    )
}

merge_shell_run_m_option <- function(option = list()) {
    if (is.null(option)) {
        option <- list()
    }
    if (!is.list(option)) {
        stop("option must be a list or NULL.", call. = FALSE)
    }

    default_option <- default_shell_run_m_option()
    unknown_names <- setdiff(names(option), names(default_option))
    if (length(unknown_names) > 0L) {
        warning(
            "Unknown shell_run_m option field(s): ",
            paste(unknown_names, collapse = ", "),
            call. = FALSE
        )
    }

    utils::modifyList(default_option, option)
}
